app/assets/javascripts/dashboard.js

```diff
@@ -127,7 +138,7 @@ MinionPoller = {
           // overview page otherwise
           if ($("#node-count").length > 0) {
             $("#node-count").text(unassignedMinions.length + " nodes found");
-          } else {
+          } else if (!hasPendingStateNode) {
```

Comment:
This causes the number of nodes to be hidden, not just the link.
![screenshot from 2017-07-20 18-26-25](https://user-images.githubusercontent.com/122395/28430678-4a3c18aa-6d79-11e7-8414-fb2be8d1eacc.png)